File: crabzilla-vertx-projector/src/test/java/io/github/crabzilla/vertx/entity/CustomerSummaryDao.java
package io.github.crabzilla.vertx.entity;

import io.github.crabzilla.example1.CustomerSummary;
import org.jdbi.v3.sqlobject.config.RegisterBeanMapper;
import org.jdbi.v3.sqlobject.customizer.Bind;
import org.jdbi.v3.sqlobject.customizer.BindBean;
import org.jdbi.v3.sqlobject.statement.SqlQuery;
import org.jdbi.v3.sqlobject.statement.SqlUpdate;

import java.util.List;

public interface CustomerSummaryDao {

  @SqlUpdate("insert into customer_summary (id, name, is_active) values " +
          "(:id, :name, false)")
  void insert(@BindBean CustomerSummary customerSummary);

  @SqlUpdate("update customer_summary set customer_summary.is_active = :isActive " +
          "where customer_summary.id = :id")
  void updateStatus(@Bind("id") String id, @Bind("isActive") Boolean isActive);

  @SqlQuery("select id, name, is_active from customer_summary")
  @RegisterBeanMapper(CustomerSummary.class)
  List<CustomerSummary> getAll();

}

File: crabzilla-kotlin-example1/crabzilla-kotlin-example1-database/src/main/java/io/github/crabzilla/example1/CustomerSummaryDao.kt
package io.github.crabzilla.example1

import org.jdbi.v3.sqlobject.statement.SqlQuery
import org.jdbi.v3.sqlobject.statement.SqlUpdate

interface CustomerSummaryDao {

  @SqlUpdate("insert into customer_summary (id, name, is_active) values " +
          "(:customerSummary.id, :customerSummary.name, false)")
  fun insert(customerSummary: CustomerSummary)

  @SqlUpdate("update customer_summary set customer_summary.is_active = :isActive " +
          "where customer_summary.id = :id")
  fun updateStatus(id: String, isActive: Boolean)

  @SqlQuery("select id, name, is_active from customer_summary")
  fun getAll(): List<CustomerSummary>

}
